build.sbt

```diff
@@ -48,7 +48,19 @@ lazy val testLibs = Seq(
   "org.scalacheck" %% "scalacheck" % scalaCheckVersion % "test"
 )
 
+lazy val gemWarts =
+  Warts.allBut(
+    Wart.Any,                // false positives
+    Wart.Nothing,            // false positives
+    Wart.Product,            // false positives
+    Wart.Serializable,       // false positives
+    Wart.Recursion,          // false positives
```

Comment:
Are there many recursion false positives ? Can `@tailrec` be used in most cases?